### PR TITLE
[Dialogue] Part 13: Bonfire of Configuration

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Dialogue is already forced everywhere, avoid giving the false impression that it is configurable.
- Avoid running an unnecessary test.
- Make steps toward removing AtlasDbHttpClients altogether. Still used in leader mode.

**Implementation Description (bullets)**:
- Remove config option for enableDialogue
- Replace switching proxies with the pure Dialogue proxy
- Remove flavors of ETE tests that used Dialogue, back when it was not default.

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Not much additional here. Existing ETEs should fail if the simplifications were wrong.

**Concerns (what feedback would you like?)**:
- This removes the ability to prefer the CJR Dialogue shim over pure Dialogue for most TimeLock communication. Is that a bad idea? We have not configured enableDialogue to false _anywhere_.

**Where should we start reviewing?**: TransactionManagers

**Priority (whenever / two weeks / yesterday)**: whenever

### CODE REVIEW

**Comment by jeremyk-91** [comment]
Probably saves us a minute on the build or so as well?

**Comment by jkozlowski** [comment]
> Probably saves us a minute on the build or so as well?

yes please

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 1]
```diff
@@ -1083,46 +1078,23 @@ private static LockAndTimestampServices getLockAndTimestampServices(
         ServiceCreator creator = ServiceCreator.withPayloadLimiter(
```
Comment (jkozlowski) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:1078]: Is this thing used still here?

Comment (jeremyk-91) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:1078]: Yeah, by `ConjureLockWatchingService`. I'll clear that out in a quick follow up to this one.

**Pull request merged**